feeds/scheduler/scheduler.go

```diff
@@ -0,0 +1,42 @@
+package scheduler
+
+import (
+	"time"
+
+	"github.com/ossf/package-feeds/feeds"
+	"github.com/ossf/package-feeds/feeds/crates"
+	"github.com/ossf/package-feeds/feeds/npm"
+	"github.com/ossf/package-feeds/feeds/pypi"
+	"github.com/ossf/package-feeds/feeds/rubygems"
+	log "github.com/sirupsen/logrus"
+)
+
+// ScheduledFeeds is a registry of feeds that should be run on a schedule
+var ScheduledFeeds = make(map[string]feeds.ScheduledFeed)
+
+func init() {
```

Comment:
How about avoiding the global state and avoiding the init functions? This will make it clean and don't have to worry about `mutex` in the future.  Instead of the global state use something like `NewObject` which returns the map.

Something like this 

```
func NewScheduler() map[string]feeds.ScheduledFeed {
	ScheduledFeeds := make(map[string]feeds.ScheduledFeed)
	ScheduledFeeds[pypi.FeedName] = pypi.Feed{}
	ScheduledFeeds[npm.FeedName] = npm.Feed{}
	ScheduledFeeds[rubygems.FeedName] = rubygems.Feed{}
	ScheduledFeeds[crates.FeedName] = crates.Feed{}
	return ScheduledFeeds
}

// PollScheduledFeeds fetches the latest packages from each registered feed
func PollScheduledFeeds(cutoff time.Time, scheduledFeeds map[string]feeds.ScheduledFeed) ([]*feeds.Package, error) {
	packages := []*feeds.Package{}
	for name, feed := range scheduledFeeds {
		logger := log.WithField("feed", name)
		pkgs, err := feed.Latest(cutoff)
		if err != nil {
			logger.WithError(err).Error("error fetching packages")
			return nil, err
		}
		processed := 0
		for _, pkg := range pkgs {
			processed++
			packages = append(packages, pkg)
		}
		logger.WithField("num_processed", processed).Print("processed packages")
	}
	return packages, nil
}
```

https://peter.bourgon.org/blog/2017/06/09/theory-of-modern-go.html
https://github.com/leighmcculloch/gochecknoinits